pkg/template/validation_test.go

```diff
@@ -0,0 +1,119 @@
+package template
+
+import (
+	"errors"
+	"reflect"
+	"strconv"
+	"testing"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+var releaseVersionsAWS1150 = Versions{
+	Calico:                       "3.10.4",
+	CRITools:                     "1.16.0",
+	Etcd:                         "3.4.9",
+	Kubernetes:                   "1.16.13",
+	KubernetesAPIHealthz:         "0.1.1",
+	KubernetesNetworkSetupDocker: "0.2.0",
+}
+
+var releaseVersionsAWS900 = Versions{
+	Calico:                       "3.9.1",
+	CRITools:                     "1.15.0",
+	Etcd:                         "3.3.15",
+	Kubernetes:                   "1.15.5",
+	KubernetesAPIHealthz:         "0.1.0",
+	KubernetesNetworkSetupDocker: "0.1.0",
+}
+
+func editVersions(versions Versions, fieldName string, version string) Versions {
+	entityType := reflect.TypeOf(versions)
```

Comment:
This function allows a field in the struct to be edited by name saving a good bit of boilerplate as we want all fields to be valid except for one in each test so we're checking the right thing.